src/aws-cpp-sdk-core/source/auth/STSCredentialsProvider.cpp

```diff
@@ -97,15 +82,15 @@ STSAssumeRoleWebIdentityCredentialsProvider::STSAssumeRoleWebIdentityCredentials
         AWS_LOGSTREAM_DEBUG(STS_ASSUME_ROLE_WEB_IDENTITY_LOG_TAG, "Resolved session_name from profile_config or environment variable to be " << m_sessionName);
     }
 
-    Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTPS;
-    config.region = tmpRegion;
 
-    Aws::Vector<Aws::String> retryableErrors;
-    retryableErrors.push_back("IDPCommunicationError");
-    retryableErrors.push_back("InvalidIdentityToken");
+    if (config.retryStrategy == nullptr) {
+        Aws::Vector<Aws::String> retryableErrors;
```

Comment:
add in place , or use reserve before emplace

Reply:
Done.